pyro/distributions/torch_distribution.py

```diff
@@ -62,6 +64,32 @@ def shape(self, sample_shape=torch.Size()):
         """
         return sample_shape + self.batch_shape + self.event_shape
 
+    def expand(self, batch_shape):
+        """
+        Expands a distribution to a desired
```

Comment:
Could you elaborate on the description here - what are the constraints on `batch_shape` provided to `.expand` and how is this different from `.expand_by`?